Source/WebCore/rendering/RenderBoxModelObject.cpp

```diff
@@ -299,40 +299,44 @@ DecodingMode RenderBoxModelObject::decodingModeForImageDraw(const Image& image,
 {
     if (!is<BitmapImage>(image))
         return DecodingMode::Synchronous;
-    
+
+    auto synchronousDecodingMode = paintInfo.context().hasPlatformContext() ? DecodingMode::Synchronous : DecodingMode::SynchronousThumbnail;
```

Comment:
We could hit this for the other "fake paints" where hasPlatformContext() is false. It might be better to plumb through a more specific "using GPU Process" flag.

Reply:
Yes this is right. Also adding SynchronousThumbnail as a new DecodingMode seems awkward and error prone. So I added the virtual method `GraphicsContext::needsNativeImageMemoryOwnership()` which returns false. I made `RemoteDisplayListRecorderProxy` override it and return true if the `renderingPurpose()` is DOM.